app/controllers/challenges_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ChallengesController < ApplicationController
+    def index
+        @challenges = Challenge.all
+    end
+
+    def show
+        @challenge = Challenge.find(params[:id])
+    end
+
+    private
+
+    def challenge_params
+        params.permit(:id, :theme, :project_budget, :project_id)
+    end
+end
```

Comment:
I'm not sure if I understand the purpose of this controller. We don't have views for this controller, and there were no requirements for views for this controller. We also didn't have a form, so I'm not sure why we need the `challenge_params` method.